### PR TITLE
newdevice: samsung printer ml series

### LINKED ISSUE #4767: Adding support for Samsung ML series printer
DO NOT DELETE THIS INFORMATION.

> Please read this information carefully.

- [X] Is your install up to date? [Updating your install](http://docs.librenms.org/General/Updating/)
      Please do not submit an issue if your install is not up to date within the last 24 hours or on a stable monthly release.
- [X] Please include all of the information between the `====================================` section of `./validate.php`.

LibreNMS Version: ead6b5b8ac7b05f36780eb6727648dc69a5d6c4d
DB Schema: 142
PHP: 7.0.11
MySQL: 10.1.17-MariaDB
RRDTool: 1.5.6
SNMP: NET-SNMP 5.7.3

- [X] If you would like us to add a new device then please provide the information asked for [here](http://docs.librenms.org/Support/FAQ/#faq20)

http://pastebin.com/m0P8EFPZ - Discovery
http://pastebin.com/tub9nE8a - Poller
http://pastebin.com/T1RmZtHu - SNMPWALk

**Please note these are AFTER applying the diff attached below.**

- [X] Please provide as much detail as possible.

Good day LibreNMS folks,

I'm trying to add support for my Samsung ML-2850ND printer.  The first part of that I think I have working nicely.  
```diff
diff --git a/includes/discovery/os/samsungprinter.inc.php b/includes/discovery/os/samsungprinter.inc.php
index 4738f20..2ae610f 100644
--- a/includes/discovery/os/samsungprinter.inc.php
+++ b/includes/discovery/os/samsungprinter.inc.php
@@ -1,5 +1,5 @@
 <?php
 
-if (str_contains($sysDescr, array('Samsung CLX', 'Samsung SCX', 'Samsung C', 'Samsung S'))) {
+if (str_contains($sysDescr, array('Samsung CLX', 'Samsung SCX', 'Samsung C', 'Samsung S', 'Samsung ML'))) {
     $os = 'samsungprinter';
 }
```

So once that lands, the device switches from a Generic device to a Samsung Printer. I get the Serial Number and the model make and it identifies the device as a printer.  However the toner graphing does not work and if I look at the output from discover / poller, it doesn't look like it's getting any of the toner info.  I'm not sure why; the SNMPwalk shows the OIDs and the data there.  So that's the part that is still a mystery to me.  

I'd be happy to provide any other information needed to help get the toner graphing up and running and device support added.

### PR DESCRIPTION
**Pull request by crcro**
#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/)
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

fix #4767

### CODE REVIEW

**Comment by Rosiak** [comment]
Hey,
tests/OSDiscoveryTest.php addition is missing.

**Comment by crcro** [comment]
added, thanks

File: tests/snmpsim/samsungprinter-ml.snmprec [hunk 1]
```diff
@@ -0,0 +1 @@
+1.3.6.1.2.1.1.1.0|4|Samsung ML-2850 Series OS 1.01.15.47 02-10-2010;Engine 1.01.25;NIC V4.01.09(ML-285x)
```
Comment (murrant) [inline tests/snmpsim/samsungprinter-ml.snmprec]: Please add the sysObjectID

Comment (crcro) [inline tests/snmpsim/samsungprinter-ml.snmprec]: done

**Pull request merged**